File: TFCycleScrollCell.m
//
//  TFCycleScrollCell.m
//  TFCycleScrollView
//
//  Created by Fengtf on 2017/12/16.
//

#import "TFCycleScrollCell.h"

@implementation TFCycleScrollCell

-(instancetype)initWithFrame:(CGRect)frame
{
    if (self = [super initWithFrame:frame]) {
        UIImageView *imageView = [[UIImageView alloc]init];
        self.imageView = imageView;
        imageView.contentMode = UIViewContentModeScaleAspectFill;
        
        UIScrollView *scrollView = [[UIScrollView alloc]init];
        self.scrollView = scrollView;
        scrollView.userInteractionEnabled = NO;
        scrollView.scrollEnabled = NO;
        
        [self.contentView addSubview:self.scrollView];
        [self.scrollView addSubview:self.imageView];
    }
    return self;
}

-(void)layoutSubviews
{
    [super layoutSubviews];
    
    self.scrollView.contentSize = self.bounds.size;
    self.scrollView.frame = self.bounds;
    self.imageView.frame = self.scrollView.bounds;
}



@end

File: TFCycleScrollView_Swift/TFCycleScrollView_Swift/TFCycleScrollView/TFCycleScrollCell.swift
//
//  TFCycleScrollCell.swift
//  Gankoo
//
//  Created by Fengtf on 2017/2/7.
//  Copyright © 2017年 ftf. All rights reserved.
//

import UIKit

class TFCycleScrollCell: UICollectionViewCell {
    
    let iconView:UIImageView = UIImageView()
    let titleLabel = UILabel()
    
    var model:TFCycleScrollModel?{
        didSet{
            guard let model = model else {
                return
            }
            let url = URL(string: model.imgUrl ?? "")
            iconView.sd_setImage(with: url, placeholderImage: nil)
        }
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
    }
    
    func initImg(){
        contentView.addSubview(iconView)
        iconView.contentMode = .scaleAspectFill
        iconView.clipsToBounds = true
    }
    
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        iconView.frame = self.bounds
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

}
